lib/ansible/modules/system/pamd.py

```diff
@@ -257,28 +298,71 @@ def __str__(self):
 # PamdService encapsulates an entire service and contains one or more rules
 class PamdService(object):
 
-    def __init__(self, path, name, ansible):
-        self.path = path
-        self.name = name
-        self.check = ansible.check_mode
+    def __init__(self, ansible=None):
+
+        if ansible is not None:
+            self.check = ansible.check_mode
+        self.check = False
         self.ansible = ansible
-        self.fname = self.path + "/" + self.name
         self.preamble = []
         self.rules = []
+        self.fname = None
+        if ansible is not None:
+            self.path = self.ansible.params["path"]
+            self.name = self.ansible.params["name"]
+
+    def load_rules_from_file(self):
+        self.fname = self.path + "/" + self.name
+        stringline = ''
         try:
             for line in open(self.fname, 'r'):
-                if line.startswith('#') and not line.isspace():
-                    self.preamble.append(line.rstrip())
-                elif not line.startswith('#') and not line.isspace():
-                    self.rules.append(PamdRule.rulefromstring
-                                      (stringline=line.rstrip()))
+                stringline += line.rstrip()
+                stringline += '\n'
+            self.load_rules_from_string(stringline)
+
         except Exception:
             e = get_exception()
-            self.ansible.fail_json(msg='Unable to open/read PAM module file ' +
-                                   '%s with error %s' % (self.fname, str(e)))
+            self.ansible.fail_json(msg='Unable to open/read PAM module \
+                                   file %s with error %s.  And line %s' %
+                                   (self.fname, str(e), stringline))
+
+    def load_rules_from_string(self, stringvalue):
+        for line in stringvalue.splitlines():
+            stringline = line.rstrip()
+            if line.startswith('#') and not line.isspace():
+                self.preamble.append(line.rstrip())
+            elif (not line.startswith('#') and
+                  not line.isspace() and
+                  len(line) != 0):
+                self.rules.append(PamdRule.rulefromstring(stringline))
+
+    def write(self):
```

Comment:
this is an unsafe way to update the file, it can lead to data corruption and other processes reading incorrect/incomplete data. write to a tmp file and use atomic_move from basic.py